From 116f6d011b32751257ed6cad420552591a88993a Mon Sep 17 00:00:00 2001
From: Uri Mashiach <uri.mashiach@compulab.co.il>
Date: Mon, 20 Jan 2020 14:41:27 +0200
Subject: [PATCH 37/43] ucm-imx8m-mini: dts: rearrange components

Change boards include relationship - the baseboard includes the module.
Move baseboard components fromthe modules to the baseboard dts file.

Signed-off-by: Uri Mashiach <uri.mashiach@compulab.co.il>
---
 arch/arm64/boot/dts/compulab/sb-ucm-imx8-rev2.dts  | 371 +++++++++++++++++++++
 arch/arm64/boot/dts/compulab/sb-ucm-imx8-rev2.dtsi |  41 ---
 .../boot/dts/compulab/ucm-imx8m-mini-dsi2.dts      |   2 +-
 .../boot/dts/compulab/ucm-imx8m-mini-lvds.dts      |   8 +-
 arch/arm64/boot/dts/compulab/ucm-imx8m-mini.dts    | 333 +-----------------
 5 files changed, 376 insertions(+), 379 deletions(-)
 create mode 100644 arch/arm64/boot/dts/compulab/sb-ucm-imx8-rev2.dts
 delete mode 100644 arch/arm64/boot/dts/compulab/sb-ucm-imx8-rev2.dtsi

diff --git a/arch/arm64/boot/dts/compulab/sb-ucm-imx8-rev2.dts b/arch/arm64/boot/dts/compulab/sb-ucm-imx8-rev2.dts
new file mode 100644
index 000000000000..a963a0d66b47
--- /dev/null
+++ b/arch/arm64/boot/dts/compulab/sb-ucm-imx8-rev2.dts
@@ -0,0 +1,371 @@
+/*
+ * Copyright (C) 2018 CompuLab Ltd.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ */
+
+#include "ucm-imx8m-mini.dts"
+
+/ {
+	model = "CompuLab SB-UCM-iMX8 board";
+	compatible = "sb-ucm-imx8", "cpl,ucm-imx8m-mini", "fsl,imx8mm";
+
+	backlight {
+		compatible = "pwm-backlight";
+		pwms = <&pwm2 0 3000000 0>;
+
+		brightness-levels = <0 1 2 4 8 16 32 64 128 255>;
+		default-brightness-level = <9>;
+		status = "okay";
+	};
+
+	simple_sound: sound {
+		compatible = "simple-audio-card";
+		simple-audio-card,name = "ucm-imx8m-mini";
+		simple-audio-card,widgets =
+			"Headphone", "Headphone Jack",
+			"Line", "Line Out",
+			"Microphone", "Mic Jack",
+			"Line", "Line In";
+		simple-audio-card,routing =
+			"Headphone Jack", "RHPOUT",
+			"Headphone Jack", "LHPOUT",
+			"MICIN", "Mic Bias",
+			"Mic Bias", "Mic Jack";
+		simple-audio-card,format = "i2s";
+		simple-audio-card,bitclock-master = <&sound_master>;
+		simple-audio-card,frame-master = <&sound_master>;
+		simple-audio-card,bitclock-inversion;
+
+		sound_master: simple-audio-card,cpu {
+			sound-dai = <&sai2>;
+			system-clock-frequency = <0>;
+			system-clock-direction = "out";
+		};
+
+		sound_codec: simple-audio-card,codec {
+			sound-dai = <&wm8731>;
+			system-clock-direction = "in";
+			system-clock-type = "mclk";
+		};
+
+	};
+
+	reg_usdhc2_vmmc: regulator-usdhc2 {
+		compatible = "regulator-fixed";
+		regulator-name = "VSD_3V3";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		gpio = <&gpio2 19 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+		startup-delay-us = <100>;
+		off-on-delay-us = <12000>;
+	};
+};
+
+&pcie0 {
+	reset-gpio = <&pca9555 0 GPIO_ACTIVE_LOW>;
+	ext_osc = <1>;
+	status = "okay";
+};
+
+&uart1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart1>;
+	assigned-clocks = <&clk IMX8MM_CLK_UART1_SRC>;
+	assigned-clock-parents = <&clk IMX8MM_SYS_PLL1_80M>;
+	status = "okay";
+};
+
+&i2c1 {
+	clock-frequency = <100000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c1>;
+	status = "okay";
+
+	pca9555:pca9555@20 {
+		compatible = "nxp,pca9555";
+		gpio-controller;
+		#gpio-cells = <2>;
+		reg = <0x20>;
+	};
+
+	eeprom@54 {
+		compatible = "atmel,24c08";
+		reg = <0x54>;
+		pagesize = <16>;
+	};
+};
+
+&i2c4 {
+	clock-frequency = <400000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c4>;
+	status = "okay";
+
+	wm8731: wm8731@1a {
+		#sound-dai-cells = <0>;
+		compatible = "wlf,wm8731";
+		reg = <0x1a>;
+		status = "okay";
+	};
+
+	ov5640_mipi: ov5640_mipi@3c {
+		compatible = "ovti,ov5640_mipi";
+		reg = <0x3c>;
+		status = "okay";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_csi1>;
+		clocks = <&clk IMX8MM_CLK_CLKO1_DIV>;
+		clock-names = "csi_mclk";
+		assigned-clocks = <&clk IMX8MM_CLK_CLKO1_SRC>,
+				  <&clk IMX8MM_CLK_CLKO1_DIV>;
+		assigned-clock-parents = <&clk IMX8MM_CLK_24M>;
+		assigned-clock-rates = <0>, <24000000>;
+		csi_id = <0>;
+		/*
+		To be looked into & fixed
+		As of now the device is alwas out of reset & under the power
+		*/
+		pwn-gpios = <&pca9555 8 GPIO_ACTIVE_LOW>;
+		rst-gpios = <&pca9555 3 GPIO_ACTIVE_HIGH>;
+
+		mclk = <24000000>;
+		mclk_source = <0>;
+		port {
+			ov5640_mipi1_ep: endpoint {
+				remote-endpoint = <&mipi1_sensor_ep>;
+			};
+		};
+	};
+	goodix_ts@5d {
+		compatible = "goodix,gt911";
+		reg = <0x5d>;
+
+		interrupt-parent = <&gpio1>;
+		interrupts = <1 IRQ_TYPE_LEVEL_HIGH>;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&touchscreen_pins>;
+
+		irq-gpios = <&gpio1 1 GPIO_ACTIVE_HIGH>;
+		reset-gpios = <&pca9555 4 GPIO_ACTIVE_HIGH>;
+		status = "okay";
+	};
+
+};
+
+&usbotg1 {
+	dr_mode = "otg";
+	status = "okay";
+};
+
+
+&usbotg2 {
+	dr_mode = "host";
+	status = "okay";
+};
+
+&usdhc2 {
+	pinctrl-names = "default", "state_100mhz", "state_200mhz";
+	pinctrl-0 = <&pinctrl_usdhc2>, <&pinctrl_usdhc2_gpio>;
+	pinctrl-1 = <&pinctrl_usdhc2_100mhz>, <&pinctrl_usdhc2_gpio>;
+	pinctrl-2 = <&pinctrl_usdhc2_200mhz>, <&pinctrl_usdhc2_gpio>;
+	cd-gpios = <&gpio2 12 GPIO_ACTIVE_LOW>;
+	wp-gpios = <&gpio2 20 GPIO_ACTIVE_HIGH>;
+	no-1-8-v;
+	bus-width = <4>;
+	vmmc-supply = <&reg_usdhc2_vmmc>;
+	status = "okay";
+};
+
+&gpu {
+	status = "okay";
+};
+
+&vpu_g1 {
+	status = "okay";
+};
+
+&vpu_g2 {
+	status = "okay";
+};
+
+&vpu_h1 {
+	status = "okay";
+};
+
+&lcdif {
+	status = "okay";
+};
+
+&mipi_dsi {
+	status = "okay";
+	panel: panel@0 {
+		compatible = "startek,kd050hdfia020";
+		reg = <0>;
+		reset-gpio = <&pca9555 6 GPIO_ACTIVE_LOW>;
+		dsi-lanes = <4>;
+		status = "okay";
+	};
+};
+
+&pwm2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pwm_backlight>;
+	status = "okay";
+};
+
+&mipi_csi_1 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "okay";
+	port {
+		mipi1_sensor_ep: endpoint1 {
+			remote-endpoint = <&ov5640_mipi1_ep>;
+			data-lanes = <2>;
+			csis-hs-settle = <13>;
+			csis-clk-settle = <2>;
+			csis-wclk;
+		};
+
+		csi1_mipi_ep: endpoint2 {
+			remote-endpoint = <&csi1_ep>;
+		};
+	};
+};
+
+&csi1_bridge {
+    fsl,mipi-mode;
+    status = "okay";
+
+    port {
+        csi1_ep: endpoint {
+            remote-endpoint = <&csi1_mipi_ep>;
+        };
+    };
+};
+
+&sai2 {
+	#sound-dai-cells = <0>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_sai2>;
+	assigned-clocks = <&clk IMX8MM_CLK_SAI2_SRC>,
+			<&clk IMX8MM_CLK_SAI2_DIV>;
+	assigned-clock-parents = <&clk IMX8MM_AUDIO_PLL1_OUT>;
+	assigned-clock-rates = <0>, <12288000>;
+	status = "okay";
+};
+
+&iomuxc {
+	sb-ucm-imx8 {
+		pinctrl_uart1: uart1grp {
+			fsl,pins = <
+				MX8MM_IOMUXC_UART1_RXD_UART1_DCE_RX	0x140
+				MX8MM_IOMUXC_UART1_TXD_UART1_DCE_TX	0x140
+			>;
+		};
+
+		pinctrl_i2c1: i2c1grp {
+			fsl,pins = <
+				MX8MM_IOMUXC_I2C1_SCL_I2C1_SCL			0x400001c3
+				MX8MM_IOMUXC_I2C1_SDA_I2C1_SDA			0x400001c3
+			>;
+		};
+
+		pinctrl_i2c4: i2c4grp {
+			fsl,pins = <
+				MX8MM_IOMUXC_I2C4_SCL_I2C4_SCL		0x400001c3
+				MX8MM_IOMUXC_I2C4_SDA_I2C4_SDA		0x400001c3
+			>;
+		};
+
+		pinctrl_usdhc2: usdhc2grp {
+			fsl,pins = <
+				MX8MM_IOMUXC_SD2_CLK_USDHC2_CLK		0x190
+				MX8MM_IOMUXC_SD2_CMD_USDHC2_CMD		0x1d0
+				MX8MM_IOMUXC_SD2_DATA0_USDHC2_DATA0	0x1d0
+				MX8MM_IOMUXC_SD2_DATA1_USDHC2_DATA1	0x1d0
+				MX8MM_IOMUXC_SD2_DATA2_USDHC2_DATA2	0x1d0
+				MX8MM_IOMUXC_SD2_DATA3_USDHC2_DATA3	0x1d0
+				MX8MM_IOMUXC_GPIO1_IO04_USDHC2_VSELECT	0x1d0
+			>;
+		};
+
+		pinctrl_usdhc2_gpio: usdhc2grpgpio {
+			fsl,pins = <
+				MX8MM_IOMUXC_SD2_RESET_B_GPIO2_IO19	0x41
+				MX8MM_IOMUXC_SD2_CD_B_GPIO2_IO12	0x41
+				MX8MM_IOMUXC_SD2_WP_GPIO2_IO20		0x00
+			>;
+		};
+
+		pinctrl_usdhc2_100mhz: usdhc2grp100mhz {
+			fsl,pins = <
+				MX8MM_IOMUXC_SD2_CLK_USDHC2_CLK		0x194
+				MX8MM_IOMUXC_SD2_CMD_USDHC2_CMD		0x1d4
+				MX8MM_IOMUXC_SD2_DATA0_USDHC2_DATA0	0x1d4
+				MX8MM_IOMUXC_SD2_DATA1_USDHC2_DATA1	0x1d4
+				MX8MM_IOMUXC_SD2_DATA2_USDHC2_DATA2	0x1d4
+				MX8MM_IOMUXC_SD2_DATA3_USDHC2_DATA3	0x1d4
+				MX8MM_IOMUXC_GPIO1_IO04_USDHC2_VSELECT	0x1d0
+			>;
+		};
+
+		pinctrl_usdhc2_200mhz: usdhc2grp200mhz {
+			fsl,pins = <
+				MX8MM_IOMUXC_SD2_CLK_USDHC2_CLK		0x196
+				MX8MM_IOMUXC_SD2_CMD_USDHC2_CMD		0x1d6
+				MX8MM_IOMUXC_SD2_DATA0_USDHC2_DATA0	0x1d6
+				MX8MM_IOMUXC_SD2_DATA1_USDHC2_DATA1	0x1d6
+				MX8MM_IOMUXC_SD2_DATA2_USDHC2_DATA2	0x1d6
+				MX8MM_IOMUXC_SD2_DATA3_USDHC2_DATA3	0x1d6
+				MX8MM_IOMUXC_GPIO1_IO04_USDHC2_VSELECT	0x1d0
+			>;
+		};
+
+		pinctrl_pwm_backlight: pwm_backlightgrp {
+			fsl,pins = <
+				MX8MM_IOMUXC_GPIO1_IO13_PWM2_OUT	0x03
+			>;
+		};
+
+		touchscreen_pins: tsgrp {
+			fsl,pins = <
+				MX8MM_IOMUXC_GPIO1_IO01_GPIO1_IO1	0x00
+			>;
+		};
+
+		/*
+		Copy & Paste from an MXP evk
+		As of now an external OSC1 25MHz is in use
+		Left as an example for the next revision
+		*/
+		pinctrl_csi1: csi1grp {
+			fsl,pins = <
+				MX8MM_IOMUXC_GPIO1_IO15_CCMSRCGPCMIX_CLKO2	0x59
+				MX8MM_IOMUXC_GPIO1_IO14_CCMSRCGPCMIX_CLKO1	0x59
+			>;
+		};
+
+		pinctrl_sai2: sai2grp {
+			fsl,pins = <
+				MX8MM_IOMUXC_SAI2_MCLK_SAI2_MCLK        0xd6
+				MX8MM_IOMUXC_SAI2_TXFS_SAI2_TX_SYNC     0xd6
+				MX8MM_IOMUXC_SAI2_RXFS_SAI2_RX_SYNC     0xd6
+				MX8MM_IOMUXC_SAI2_TXC_SAI2_TX_BCLK      0xd6
+				MX8MM_IOMUXC_SAI2_TXD0_SAI2_TX_DATA0    0xd6
+				MX8MM_IOMUXC_SAI2_RXD0_SAI2_RX_DATA0    0xd6
+			>;
+		};
+
+	};
+};
diff --git a/arch/arm64/boot/dts/compulab/sb-ucm-imx8-rev2.dtsi b/arch/arm64/boot/dts/compulab/sb-ucm-imx8-rev2.dtsi
deleted file mode 100644
index e306b1f3adb2..000000000000
--- a/arch/arm64/boot/dts/compulab/sb-ucm-imx8-rev2.dtsi
+++ /dev/null
@@ -1,41 +0,0 @@
-/*
- * Copyright (C) 2018 CompuLab Ltd.
- *
- * This program is free software; you can redistribute it and/or
- * modify it under the terms of the GNU General Public License
- * as published by the Free Software Foundation; either version 2
- * of the License, or (at your option) any later version.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
- * GNU General Public License for more details.
- */
-
-#include <dt-bindings/pinctrl/pins-imx8mm.h>
-
-&pcie0 {
-	reset-gpio = <&pca9555 0 GPIO_ACTIVE_LOW>;
-	ext_osc = <1>;
-	status = "okay";
-};
-
-&i2c1 {
-	clock-frequency = <100000>;
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_i2c1>;
-	status = "okay";
-
-	pca9555:pca9555@20 {
-		compatible = "nxp,pca9555";
-		gpio-controller;
-		#gpio-cells = <2>;
-		reg = <0x20>;
-	};
-
-	eeprom@54 {
-		compatible = "atmel,24c08";
-		reg = <0x54>;
-		pagesize = <16>;
-	};
-};
diff --git a/arch/arm64/boot/dts/compulab/ucm-imx8m-mini-dsi2.dts b/arch/arm64/boot/dts/compulab/ucm-imx8m-mini-dsi2.dts
index 27dc899c9b88..49a8c3f7458d 100644
--- a/arch/arm64/boot/dts/compulab/ucm-imx8m-mini-dsi2.dts
+++ b/arch/arm64/boot/dts/compulab/ucm-imx8m-mini-dsi2.dts
@@ -12,7 +12,7 @@
  * GNU General Public License for more details.
  */
 
-#include "ucm-imx8m-mini.dts"
+#include "sb-ucm-imx8-rev2.dts"
 
 &panel {
 	dsi-lanes = <2>;
diff --git a/arch/arm64/boot/dts/compulab/ucm-imx8m-mini-lvds.dts b/arch/arm64/boot/dts/compulab/ucm-imx8m-mini-lvds.dts
index e9511ef185f5..72aa389e7f31 100644
--- a/arch/arm64/boot/dts/compulab/ucm-imx8m-mini-lvds.dts
+++ b/arch/arm64/boot/dts/compulab/ucm-imx8m-mini-lvds.dts
@@ -12,13 +12,11 @@
  * GNU General Public License for more details.
  */
 
-/dts-v1/;
-
-#include "ucm-imx8m-mini.dts"
+#include "sb-ucm-imx8-rev2.dts"
 
 / {
-    model = "CompuLab UCM i.MX8MM board (lvds)";
-    compatible = "cpl,ucm-imx8m-mini", "fsl,imx8mm";
+    model = "CompuLab EB DSI2LVDS board (lvds)";
+    compatible = "eb-dsi2lvds", "sb-ucm-imx8", "cpl,ucm-imx8m-mini", "fsl,imx8mm";
 };
 
 &iomuxc {
diff --git a/arch/arm64/boot/dts/compulab/ucm-imx8m-mini.dts b/arch/arm64/boot/dts/compulab/ucm-imx8m-mini.dts
index 3a4aacfa95c9..92ddb0bed3d0 100644
--- a/arch/arm64/boot/dts/compulab/ucm-imx8m-mini.dts
+++ b/arch/arm64/boot/dts/compulab/ucm-imx8m-mini.dts
@@ -14,8 +14,8 @@
 
 /dts-v1/;
 
+#include <dt-bindings/pinctrl/pins-imx8mm.h>
 #include "../freescale/fsl-imx8mm.dtsi"
-#include "sb-ucm-imx8-rev2.dtsi"
 
 / {
 	model = "CompuLab UCM i.MX8MM board";
@@ -50,17 +50,6 @@
 		status = "okay";
 	};
 
-	reg_usdhc2_vmmc: regulator-usdhc2 {
-		compatible = "regulator-fixed";
-		regulator-name = "VSD_3V3";
-		regulator-min-microvolt = <3300000>;
-		regulator-max-microvolt = <3300000>;
-		gpio = <&gpio2 19 GPIO_ACTIVE_HIGH>;
-		enable-active-high;
-		startup-delay-us = <100>;
-		off-on-delay-us = <12000>;
-	};
-
 	regulator-usdhc3rst {
 		compatible = "regulator-fixed";
 		regulator-name = "usdhc3_rst";
@@ -117,46 +106,6 @@
 		vin-supply = <&regulator_bt_reg_on>;
 	};
 
-	simple_sound: sound {
-		compatible = "simple-audio-card";
-		simple-audio-card,name = "ucm-imx8m-mini";
-		simple-audio-card,widgets =
-			"Headphone", "Headphone Jack",
-			"Line", "Line Out",
-			"Microphone", "Mic Jack",
-			"Line", "Line In";
-		simple-audio-card,routing =
-			"Headphone Jack", "RHPOUT",
-			"Headphone Jack", "LHPOUT",
-			"MICIN", "Mic Bias",
-			"Mic Bias", "Mic Jack";
-		simple-audio-card,format = "i2s";
-		simple-audio-card,bitclock-master = <&sound_master>;
-		simple-audio-card,frame-master = <&sound_master>;
-		simple-audio-card,bitclock-inversion;
-
-		sound_master: simple-audio-card,cpu {
-			sound-dai = <&sai2>;
-			system-clock-frequency = <0>;
-			system-clock-direction = "out";
-		};
-
-		sound_codec: simple-audio-card,codec {
-			sound-dai = <&wm8731>;
-			system-clock-direction = "in";
-			system-clock-type = "mclk";
-		};
-
-	};
-
-	backlight {
-		compatible = "pwm-backlight";
-		pwms = <&pwm2 0 3000000 0>;
-
-		brightness-levels = <0 1 2 4 8 16 32 64 128 255>;
-		default-brightness-level = <9>;
-		status = "okay";
-	};
 };
 
 &clk {
@@ -205,14 +154,6 @@
 			>;
 		};
 
-
-		pinctrl_i2c1: i2c1grp {
-			fsl,pins = <
-				MX8MM_IOMUXC_I2C1_SCL_I2C1_SCL			0x400001c3
-				MX8MM_IOMUXC_I2C1_SDA_I2C1_SDA			0x400001c3
-			>;
-		};
-
 		pinctrl_i2c2: i2c2grp {
 			fsl,pins = <
 				MX8MM_IOMUXC_I2C2_SCL_I2C2_SCL			0x400001c3
@@ -220,33 +161,12 @@
 			>;
 		};
 
-		pinctrl_i2c3: i2c3grp {
-			fsl,pins = <
-				MX8MM_IOMUXC_I2C3_SCL_I2C3_SCL			0x400001c3
-				MX8MM_IOMUXC_I2C3_SDA_I2C3_SDA			0x400001c3
-			>;
-		};
-
-		pinctrl_i2c4: i2c4grp {
-			fsl,pins = <
-				MX8MM_IOMUXC_I2C4_SCL_I2C4_SCL			0x400001c3
-				MX8MM_IOMUXC_I2C4_SDA_I2C4_SDA			0x400001c3
-			>;
-		};
-
 		pinctrl_pmic: pmicirq {
 			fsl,pins = <
 				MX8MM_IOMUXC_GPIO1_IO03_GPIO1_IO3		0x41
 			>;
 		};
 
-		pinctrl_uart1: uart1grp {
-			fsl,pins = <
-				MX8MM_IOMUXC_UART1_RXD_UART1_DCE_RX	0x140
-				MX8MM_IOMUXC_UART1_TXD_UART1_DCE_TX	0x140
-			>;
-		};
-
 		pinctrl_uart3: uart3grp {
 			fsl,pins = <
 				MX8MM_IOMUXC_UART3_RXD_UART3_DCE_RX	0x49
@@ -306,50 +226,6 @@
 		};
 
 
-		pinctrl_usdhc2_gpio: usdhc2grpgpio {
-			fsl,pins = <
-				MX8MM_IOMUXC_SD2_RESET_B_GPIO2_IO19	0x41
-				MX8MM_IOMUXC_SD2_CD_B_GPIO2_IO12	0x41
-				MX8MM_IOMUXC_SD2_WP_GPIO2_IO20		0x00
-			>;
-		};
-
-		pinctrl_usdhc2: usdhc2grp {
-			fsl,pins = <
-				MX8MM_IOMUXC_SD2_CLK_USDHC2_CLK		0x190
-				MX8MM_IOMUXC_SD2_CMD_USDHC2_CMD		0x1d0
-				MX8MM_IOMUXC_SD2_DATA0_USDHC2_DATA0	0x1d0
-				MX8MM_IOMUXC_SD2_DATA1_USDHC2_DATA1	0x1d0
-				MX8MM_IOMUXC_SD2_DATA2_USDHC2_DATA2	0x1d0
-				MX8MM_IOMUXC_SD2_DATA3_USDHC2_DATA3	0x1d0
-				MX8MM_IOMUXC_GPIO1_IO04_USDHC2_VSELECT	0x1d0
-			>;
-		};
-
-		pinctrl_usdhc2_100mhz: usdhc2grp100mhz {
-			fsl,pins = <
-				MX8MM_IOMUXC_SD2_CLK_USDHC2_CLK		0x194
-				MX8MM_IOMUXC_SD2_CMD_USDHC2_CMD		0x1d4
-				MX8MM_IOMUXC_SD2_DATA0_USDHC2_DATA0	0x1d4
-				MX8MM_IOMUXC_SD2_DATA1_USDHC2_DATA1	0x1d4
-				MX8MM_IOMUXC_SD2_DATA2_USDHC2_DATA2	0x1d4
-				MX8MM_IOMUXC_SD2_DATA3_USDHC2_DATA3	0x1d4
-				MX8MM_IOMUXC_GPIO1_IO04_USDHC2_VSELECT	0x1d0
-			>;
-		};
-
-		pinctrl_usdhc2_200mhz: usdhc2grp200mhz {
-			fsl,pins = <
-				MX8MM_IOMUXC_SD2_CLK_USDHC2_CLK		0x196
-				MX8MM_IOMUXC_SD2_CMD_USDHC2_CMD		0x1d6
-				MX8MM_IOMUXC_SD2_DATA0_USDHC2_DATA0	0x1d6
-				MX8MM_IOMUXC_SD2_DATA1_USDHC2_DATA1	0x1d6
-				MX8MM_IOMUXC_SD2_DATA2_USDHC2_DATA2	0x1d6
-				MX8MM_IOMUXC_SD2_DATA3_USDHC2_DATA3	0x1d6
-				MX8MM_IOMUXC_GPIO1_IO04_USDHC2_VSELECT	0x1d0
-			>;
-		};
-
 		pinctrl_usdhc3: usdhc3grp {
 			fsl,pins = <
 				MX8MM_IOMUXC_NAND_WE_B_USDHC3_CLK		0x40000190
@@ -403,40 +279,6 @@
 				MX8MM_IOMUXC_GPIO1_IO02_WDOG1_WDOG_B		0xc6
 			>;
 		};
-
-		touchscreen_pins: tsgrp {
-			fsl,pins = <
-				MX8MM_IOMUXC_GPIO1_IO01_GPIO1_IO1		0x00
-			>;
-		};
-		/*
-		Copy & Paste from an MXP evk
-		As of now an external OSC1 25MHz is in use
-		Left as an example for the next revision
-		*/
-		pinctrl_csi1: csi1grp {
-			fsl,pins = <
-				MX8MM_IOMUXC_GPIO1_IO15_CCMSRCGPCMIX_CLKO2	0x59
-				MX8MM_IOMUXC_GPIO1_IO14_CCMSRCGPCMIX_CLKO1	0x59
-			>;
-		};
-
-		pinctrl_sai2: sai2grp {
-			fsl,pins = <
-				MX8MM_IOMUXC_SAI2_MCLK_SAI2_MCLK        0xd6
-				MX8MM_IOMUXC_SAI2_TXFS_SAI2_TX_SYNC     0xd6
-				MX8MM_IOMUXC_SAI2_RXFS_SAI2_RX_SYNC     0xd6
-				MX8MM_IOMUXC_SAI2_TXC_SAI2_TX_BCLK      0xd6
-				MX8MM_IOMUXC_SAI2_TXD0_SAI2_TX_DATA0    0xd6
-				MX8MM_IOMUXC_SAI2_RXD0_SAI2_RX_DATA0    0xd6
-			>;
-		};
-
-		pinctrl_pwm_backlight: pwm_backlightgrp {
-			fsl,pins = <
-				MX8MM_IOMUXC_GPIO1_IO13_PWM2_OUT	0x03
-			>;
-		};
 	};
 };
 
@@ -610,101 +452,6 @@
 	};
 };
 
-&i2c3 {
-	clock-frequency = <100000>;
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_i2c3>;
-	status = "disabled";
-};
-
-&i2c4 {
-	clock-frequency = <400000>;
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_i2c4>;
-	status = "okay";
-
-	wm8731: wm8731@1a {
-		#sound-dai-cells = <0>;
-		compatible = "wlf,wm8731";
-		reg = <0x1a>;
-		status = "okay";
-	};
-
-	ov5640_mipi: ov5640_mipi@3c {
-		compatible = "ovti,ov5640_mipi";
-		reg = <0x3c>;
-		status = "okay";
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_csi1>;
-		clocks = <&clk IMX8MM_CLK_CLKO1_DIV>;
-		clock-names = "csi_mclk";
-		assigned-clocks = <&clk IMX8MM_CLK_CLKO1_SRC>,
-				  <&clk IMX8MM_CLK_CLKO1_DIV>;
-		assigned-clock-parents = <&clk IMX8MM_CLK_24M>;
-		assigned-clock-rates = <0>, <24000000>;
-		csi_id = <0>;
-		/*
-		To be looked into & fixed
-		As of now the device is alwas out of reset & under the power
-		*/
-		pwn-gpios = <&pca9555 8 GPIO_ACTIVE_LOW>;
-		rst-gpios = <&pca9555 3 GPIO_ACTIVE_HIGH>;
-
-		mclk = <24000000>;
-		mclk_source = <0>;
-		port {
-			ov5640_mipi1_ep: endpoint {
-				remote-endpoint = <&mipi1_sensor_ep>;
-			};
-		};
-	};
-	goodix_ts@5d {
-		compatible = "goodix,gt911";
-		reg = <0x5d>;
-
-		interrupt-parent = <&gpio1>;
-		interrupts = <1 IRQ_TYPE_LEVEL_HIGH>;
-
-		pinctrl-names = "default";
-		pinctrl-0 = <&touchscreen_pins>;
-
-		irq-gpios = <&gpio1 1 GPIO_ACTIVE_HIGH>;
-		reset-gpios = <&pca9555 4 GPIO_ACTIVE_HIGH>;
-		status = "okay";
-	};
-
-};
-
-&mipi_csi_1 {
-	#address-cells = <1>;
-	#size-cells = <0>;
-	status = "okay";
-	port {
-		mipi1_sensor_ep: endpoint1 {
-			remote-endpoint = <&ov5640_mipi1_ep>;
-			data-lanes = <2>;
-			csis-hs-settle = <13>;
-			csis-clk-settle = <2>;
-			csis-wclk;
-		};
-
-		csi1_mipi_ep: endpoint2 {
-			remote-endpoint = <&csi1_ep>;
-		};
-	};
-};
-
-&csi1_bridge {
-    fsl,mipi-mode;
-    status = "okay";
-
-    port {
-        csi1_ep: endpoint {
-            remote-endpoint = <&csi1_mipi_ep>;
-        };
-    };
-};
-
 &snvs {
 	status = "okay";
 };
@@ -732,14 +479,6 @@
 	};
 };
 
-&uart1 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_uart1>;
-	assigned-clocks = <&clk IMX8MM_CLK_UART1_SRC>;
-	assigned-clock-parents = <&clk IMX8MM_SYS_PLL1_80M>;
-	status = "okay";
-};
-
 &uart3 { /* console */
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_uart3>;
@@ -771,18 +510,6 @@
 	};
 };
 
-&usdhc2 {
-	pinctrl-names = "default", "state_100mhz", "state_200mhz";
-	pinctrl-0 = <&pinctrl_usdhc2>, <&pinctrl_usdhc2_gpio>;
-	pinctrl-1 = <&pinctrl_usdhc2_100mhz>, <&pinctrl_usdhc2_gpio>;
-	pinctrl-2 = <&pinctrl_usdhc2_200mhz>, <&pinctrl_usdhc2_gpio>;
-	cd-gpios = <&gpio2 12 GPIO_ACTIVE_LOW>;
-	wp-gpios = <&gpio2 20 GPIO_ACTIVE_HIGH>;
-	no-1-8-v;
-	bus-width = <4>;
-	vmmc-supply = <&reg_usdhc2_vmmc>;
-	status = "okay";
-};
 
 &usdhc3 {
 	compatible = "fsl,imx8mq-usdhc\0fsl,imx7d-usdhc";
@@ -808,64 +535,6 @@
 	arm-supply = <&buck2_reg>;
 };
 
-&gpu {
-	status = "okay";
-};
-
-&vpu_g1 {
-	status = "okay";
-};
-
-&vpu_g2 {
-	status = "okay";
-};
-
-&vpu_h1 {
-	status = "okay";
-};
-
-&usbotg1 {
-	dr_mode = "otg";
-	status = "okay";
-};
-
-&usbotg2 {
-	dr_mode = "host";
-	status = "okay";
-};
-
-&sai2 {
-	#sound-dai-cells = <0>;
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_sai2>;
-	assigned-clocks = <&clk IMX8MM_CLK_SAI2_SRC>,
-			<&clk IMX8MM_CLK_SAI2_DIV>;
-	assigned-clock-parents = <&clk IMX8MM_AUDIO_PLL1_OUT>;
-	assigned-clock-rates = <0>, <12288000>;
-	status = "okay";
-};
-
-&lcdif {
-	status = "okay";
-};
-
-&pwm2 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_pwm_backlight>;
-	status = "okay";
-};
-
-&mipi_dsi {
-	status = "okay";
-	panel: panel@0 {
-		compatible = "startek,kd050hdfia020";
-		reg = <0>;
-		reset-gpio = <&pca9555 6 GPIO_ACTIVE_LOW>;
-		dsi-lanes = <4>;
-		status = "okay";
-	};
-};
-
 &cpu_alert0 {
 	temperature = <105000>;
 };
-- 
2.11.0

